### DISCUSSION TITLE
dired unusable "no file on this line"

### DISCUSSION THREAD

**Discussion by Inc0n** [d12frosted/homebrew-emacs-plus#394] [Q&A] [answered]
I am not sure what may be the cause of this issue, but it I had this one ever since emacs plus installed.

Pressing `ret` (`dired-find-file`) will prompt this.
Pressing `R` (`dired-do-rename`) will prompt "No file specified".

Basically dired is unusable.

I would like some help on this one, any fixes? Is anyone else experiencing this issue as well?

**Comment by Inc0n** [comment, marked as answer]
I just found out in https://emacsredux.com/blog/2015/05/09/emacs-on-os-x/

> OS X ships with BSD’s ls command which doesn’t have all the features of GNU ls (used internally by dired). Fortunately, this is easily solvable:
> ```
> $ brew install coreutils
> ```
> To avoid conflicts the GNU utils are prefixed with g, so ls becomes gls.
> ```
> (setq insert-directory-program (executable-find "gls"))
> ```

**Reply by d12frosted** [reply]
Glad you figured that out!
